butterfree/load/sink.py

```diff
@@ -101,8 +109,33 @@ def flush(
 
         self.validation.input(pre_hook_df).check()
 
+        for writer in self.writers:
+            if isinstance(writer, HistoricalFeatureStoreWriter):
+                writer.add_pre_hook(
+                    SparkTableSchemaCompatibilityHook(
+                        spark_client, f"{feature_set.name}", f"{writer.database}"
+                    )
+                )
+            if isinstance(writer, OnlineFeatureStoreWriter):
+                table_name = (
+                    feature_set.entity if writer.write_to_entity else feature_set.name
+                )
+
+                cassandra_client = CassandraClient(
+                    cassandra_host=writer.db_config.host,
+                    cassandra_keyspace=writer.db_config.keyspace,
+                    cassandra_user=writer.db_config.username,
+                    cassandra_password=writer.db_config.password,
+                )
+
+                writer.add_pre_hook(
+                    CassandraTableSchemaCompatibilityHook(
+                        cassandra_client, f"{table_name}"
+                    )
+                )
```

Comment:
So this is a proposal, to be able to fit the hooks before calling the writer, we would not be able to put it in the constructor because we would not have all the necessary parameters. Another idea I had was to add Hook in `db_config` because they are already specific.

Reply:
Good point here! So I think this must be discussed better with the team. Maybe the initial interfaces thought for the `Hook` and for schema compatibility hooks (Spark and Cassandra) are not fitting very well. Table names are only available in the context of the method `run` 🤔

Reply:
I agree with @rafaelleinio, the sink should know nothing about the writers. Also, the writers should know nothing about the system they are writing to, so the HistoricalFeatureStore doesn't mean SparkMetastore and OnlineFeatureStore doesn't mean Cassandra, that's why they all are defined with a `db_config` parameter. If there are required hooks for a specific db, they should be implemented inside that class. Being agnostic means we can other writers in the sink and also means users can choose other database systems to write into.